rules/manager.go

```diff
@@ -747,6 +753,62 @@ func (m *Manager) loadGroups(interval time.Duration, filenames ...string) (map[s
 	return groups, nil
 }
 
+// testTemplateParsing checks if the templates used in labels and annotations
+// of the alerting rules are parsed correctly.
+// NOTE: It wont detect if a wrong key is used with the variables, like `$labels.wrongKey`.
+//       This will check for wrong variables used, like `$wrongLabelsVariable`.
+//       Also wrong usage of template like {{$labels.quantile * 100}}, etc.
+func testTemplateParsing(ctx context.Context, queryFunc QueryFunc, rgs *rulefmt.RuleGroups) (errs []error) {
+	for _, g := range rgs.Groups {
+		for _, rl := range g.Rules {
+			if rl.Alert == "" {
+				// Not an alerting rule.
+				continue
+			}
+
+			// Trying to parse templates.
+			tmplData := struct {
+				Labels map[string]string
+				Value  float64
+			}{
+				Labels: make(map[string]string),
+			}
+			defs := "{{$labels := .Labels}}{{$value := .Value}}"
+			parseTest := func(text string) error {
+				tmpl := template.NewTemplateExpander(
+					ctx,
+					defs+text,
+					"__alert_"+rl.Alert,
+					tmplData,
+					model.Time(timestamp.FromTime(time.Now())),
+					template.QueryFunc(queryFunc),
+					nil,
+				)
+				err := tmpl.ParseTest()
+				return err
```

Comment:
`return tmpl.ParseTest()`

Reply:
Further, can we print which alert is erroring so that it is easy to debug and fix for the user?